### ISSUE TITLE
osxapp build fails to link

### ISSUE THREAD

**Issue by jfirebaugh** [mapbox/mapbox-gl-native#4739]
```
Undefined symbols for architecture x86_64:
  "_SCErrorString", referenced from:
      -[MGLReachability startNotifier] in libmbgl-platform-osx.a(reachability.o)
  "_SCNetworkReachabilityCreateWithAddress", referenced from:
      +[MGLReachability reachabilityWithAddress:] in libmbgl-platform-osx.a(reachability.o)
  "_SCNetworkReachabilityCreateWithName", referenced from:
      +[MGLReachability reachabilityWithHostname:] in libmbgl-platform-osx.a(reachability.o)
  "_SCNetworkReachabilityGetFlags", referenced from:
      -[MGLReachability isReachable] in libmbgl-platform-osx.a(reachability.o)
      -[MGLReachability isReachableViaWiFi] in libmbgl-platform-osx.a(reachability.o)
      -[MGLReachability connectionRequired] in libmbgl-platform-osx.a(reachability.o)
      -[MGLReachability isConnectionOnDemand] in libmbgl-platform-osx.a(reachability.o)
      -[MGLReachability isInterventionRequired] in libmbgl-platform-osx.a(reachability.o)
      -[MGLReachability reachabilityFlags] in libmbgl-platform-osx.a(reachability.o)
  "_SCNetworkReachabilitySetCallback", referenced from:
      -[MGLReachability startNotifier] in libmbgl-platform-osx.a(reachability.o)
      -[MGLReachability stopNotifier] in libmbgl-platform-osx.a(reachability.o)
  "_SCNetworkReachabilitySetDispatchQueue", referenced from:
      -[MGLReachability startNotifier] in libmbgl-platform-osx.a(reachability.o)
      -[MGLReachability stopNotifier] in libmbgl-platform-osx.a(reachability.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

cc @1ec5

**Comment by 1ec5** [comment]
I can’t reproduce these errors, either in Xcode or on the command line `make osx`. What target triggered these warnings?

**Comment by jfirebaugh** [comment]
oxapp target, Xcode 7.2.

**Comment by 1ec5** [comment]
Same, but it’s working for me. What happens if you hold down Option and go to Product ‣ Clean Build Folder – basically the same as deleting the DerivedData folder – then rebuild?

**Comment by jfirebaugh** [comment]
Same result.

**Comment by jfirebaugh** [comment]
I see that linking to `SystemConfiguration.framework` was removed from platform.gyp in 15769537c308560eaa7f467fca9302f43d518ca6. Was this moved somewhere else?

**Comment by 1ec5** [comment]
As a system framework, it should be linked automatically due to the import statement in reachability.h and `LINK_WITH_STANDARD_LIBRARIES=YES`. But it couldn’t hurt to explicitly link it. Generally speaking, I tried to have only the applications and dynamic frameworks link to system frameworks.

**Comment by jfirebaugh** [comment]
Fixed in 0c44266ffb1bba02e2d6034f28b5ea8739bdde70.
